src/UnderlineNav2/UnderlineNav.test.tsx

```diff
@@ -119,3 +146,55 @@ describe('UnderlineNav', () => {
     expect(heading.textContent).toBe('Repository navigation')
   })
 })
+
+describe('Keyboard Navigation', () => {
+  it('should move focus to the next/previous item on the list with horizontal arrow keys', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
+    const nextItem = getByText('Issues').closest('a') as HTMLAnchorElement
+    // Focus first item
+    item.focus()
+    // Press right arrow
+    fireEvent.keyDown(item, arrowRight)
+    // focus should be on the next item
+    expect(nextItem).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('0')
+    fireEvent.keyDown(nextItem, arrowLeft)
+    // focus should be on the previous item
+    expect(item).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('-1')
+    expect(item.getAttribute('tabindex')).toBe('0')
+  })
+  it('should move focus to the next/previous item on the list with vertical arrow keys', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
+    const nextItem = getByText('Issues').closest('a') as HTMLAnchorElement
+    // Focus first item
+    item.focus()
+    // Press down arrow
+    fireEvent.keyDown(item, arrowDown)
+    // focus should be on the next item
+    expect(nextItem).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('0')
+    // Press up arrow
+    fireEvent.keyDown(nextItem, arrowUp)
+    // focus should be on the previous item
+    expect(item).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('-1')
+    expect(item.getAttribute('tabindex')).toBe('0')
+  })
+  it('should move focus to the next/previous item on the list with the tab key', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
+    const nextItem = getByText('Issues').closest('a') as HTMLAnchorElement
+    // Focus first item
+    item.focus()
+    // Press down arrow
+    fireEvent.keyDown(item, tab)
```

Comment:
When do you think would be best to use `fireEvent` versus `user.tab()` from user events? e.g. https://testing-library.com/docs/user-event/convenience/#tab

Reply:
I didn't really know the differences! 😿 Thanks for pointing this out! I got the habit of using fireEvent but I just read about both now and using `userEvents` makes better sense here because we want to simulate the exact behaviour of user pressing the `tab` key rather than just firing the actual keydown event. TIL that userEvents might include some other chain of events before the actual keydown fireEvent. 

To answer your question, using `fireEvent` might be a better use when probably we want to test the actually event in an isolated way or if we need a different set of chain events before firing a keydown event. For example if we want to test tab key firing without focusing it maybe